unity/Assets/Scripts/ArtGallery/Util/ILighthouseToLightHouseVisibility.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine;
+using Util.Geometry.Polygon;
+
+namespace Assets.Scripts.ArtGallery.Util
+{
+    public interface ILighthouseToLightHouseVisibility

```

Comment:
Does this interface really need all of these methods or could a subset of them also suffice? Now if someone in the future wants to add yet another visibility algorithm, they would need to implement all of these.

Reply:
There are 5 currently. 
TBH 1 should be enough. Given this list of vertices and the polygon can everything in this list be seen from at least one other vertex in the list. 

The other methods are more of a build up. They can be handy if you want to check if a single vertex can be seen, or if two vertexes can see each other. 

The other two are just helper functions that if you are smart enough to be at uni someone should be able to make quickly. Ill remove the helper functions.